auth0/provider.go

```diff
@@ -88,15 +88,15 @@ func Provider() *schema.Provider {
 		},
 	}
 
-	provider.ConfigureContextFunc = ConfigureProvider(provider.TerraformVersion)
+	provider.ConfigureContextFunc = configureProvider(&provider.TerraformVersion)
```

Comment:
We need to send the pointer to the terraform version as otherwise this will get evaluated when building the binary instead of getting evaluated before making the request leaving us with an empty terraform version inside the user agent.